design/community/book-proposal.md

```diff
@@ -0,0 +1,185 @@
+
+2. Configure Netlify for the book preview as part of a PR review for document changes.
+
+3. Write-down and add missing documents.
+
+### Non-Goals
+
+1. Create developer-guide
+
+2. Keep HTML/CSS files within the repository
+
+## Proposal
+
+Use mdbook tool to build the user-guide for the Metal³ project, which
+will combine documents under metal3-docs/docs and publish them as a single user-guide.
+
+### User Stories
+
+* As the Metal³ user I would like to have a document that would help me to
+   get started with the project easily.
+
+## Design Details
+
+### Why mdbook
+
+1. Renders a book content from markdown files
+
+2. Automatically generates all the HTML/CSS files for you
+
+3. Used by several open source projects including Kubernetes ecosystems such as
+    Cluster API, Cluster-api-provider-aws, some docs of Kubernetes
+    and etc.
+
+4. Offers preview capabilities to detect issues before they get merged.
+
+Netlify deployment will be configured for the metal3-docs repository to have a
+preview process when a future commit introduces documentation changes.
+
+**How do we keep the user-guide in sync with documentation changes in the
+    metal3-docs repository?**
+
+There are two possible ways for that.
+
+1. Have a Makefile target/script in the Metal3-docs repository to build the user-guide
+    from the current content. And configure Netlify automatic-triggering which will
+    first run the Makefile target/script under Metal3-docs and then publish the output
+    (i.e. final user-guide with updated content).
+
+2. Add the script within Netlify to build the user-guide from the current content
+    and configure Netlify automatic-triggering.
+
+**Where do we store mdbook auto-generated files like CSS/HTML ?**
+
+Since we are using Netlify, we don’t have to store those files within the Metal3-docs
+Github repository. Because, Netlify will run the mdbook binary, which will be looking
+for the book structure file in `.toml` format. As such, the user-guide is generated
+on the fly, and the output (i.e. online user-guide) is stored and published via
+the Netlify URL.
+
+**How will the book preview work when there is a PR with doc changes in the
+    Metal3-docs repository?**
+
+Netlify automatically gets triggered on every PR opened in the Metal3-docs repository.
+And after that, it will follow either step1 or step2 (in question *How do we keep
+the user-guide in sync with documentation changes in the metal3-docs repository?*)
+and publish the final-user guide and a temporary URL.
+
+**How often do we trigger the Netlify build for the final book?**
+
+Netlify allows us to configure automatic-triggering N times per day. As such,
+we can decide how many times in a day we want to trigger it. And it actually
+depends on what’s the frequency of PRs in the Metal3-docs repository. But since
+the frequency of the documentation is very low we can configure it to be once a
+day for now.
```

Comment:
We can only configure periodic jobs?

Reply:
Sorry if I wasn't clear enough.  Basically, Netlify will be triggered in two different situations.

1.  Auto build for updating the published user- guide. This can be configured to be run/triggered as much as we want in a day/week. This is just to generate the latest user-guide from the repo. For example, if there is a PR that got merged, we won't see the same change immediately in the published user-guide, until the Netlify auto build gets triggered. But for example, in the next day we for sure will have the same changes in the final user-guide.
2. Automatic triggering on every PR for the preview. This is actually doesn't require any configuration from us. It's just that we tell to Netlify which repo it should watch. And the rest it will handle itself. 

I'm not sure if it is possible that Netlify build can also be triggered after the PR gets merged. If that's possible, then we don't have to even configure periodic triggering. I can check that from Netlify.

Reply:
OK, it was the latter that I was asking about. If we can only publish periodically, I think daily is an acceptable frequency.